bin/varnishd/cache/cache_hash.c

```diff
@@ -921,6 +921,39 @@ HSH_Cancel(struct worker *wrk, struct objcore *oc, struct boc *boc)
 	ObjSlim(wrk, oc);
 }
 
+/*---------------------------------------------------------------------
+ * Abandon an objcore that will not proceed with a fetch.
+ */
+
+int
+HSH_Abandon(struct worker *wrk, struct objcore **ocp)
+{
+	struct objhead *oh;
+	struct objcore *oc;
+	struct rush rush;
+
+	CHECK_OBJ_NOTNULL(wrk, WORKER_MAGIC);
+	TAKE_OBJ_NOTNULL(oc, ocp, OBJCORE_MAGIC);
+	INIT_OBJ(&rush, RUSH_MAGIC);
+
+	oh = oc->objhead;
+	CHECK_OBJ(oh, OBJHEAD_MAGIC);
+
+	Lck_Lock(&oh->mtx);
```

Comment:
Take the `oh->mtx` a second time and duplicate the rush code for _what_ exactly?

Reply:
What do you mean by second time?

The point here is to unbusy the objcore on behalf of the nonexistent fetch task that would normally have ensured it.

Reply:
I'm finally seeing the double lock problem, which happens below the code referenced by your comment.

Addressed in ea4ba6f2295fdba354723109e473cab80bf643dd.